packages/lib/src/vis/tiles/TiledHeatmapLayer.tsx

```diff
@@ -0,0 +1,61 @@
+import { Suspense } from 'react';
+import { LinearFilter, NearestFilter, Vector2 } from 'three';
+
+import { useVisibleDomains } from '../hooks';
+import Tile from './Tile';
+import type { TilesApi } from './api';
+import type { ColorMapAttributes } from './models';
+import { getTilesOffset, tileToCenterSquareDistance } from './utils';
+
+interface Props extends ColorMapAttributes {
+  api: TilesApi;
+  lod: number;
+}
+
+function TiledHeatmapLayer(props: Props) {
+  const { api, lod, ...colorMapAttrs } = props;
+
+  const scale = 2 ** lod;
+  const { width, height } = api.levelsSize[lod];
+  const { xVisibleDomain, yVisibleDomain } = useVisibleDomains();
+
+  const xOrigin = Math.max(0, xVisibleDomain[0] / scale);
+  const yOrigin = Math.max(0, yVisibleDomain[0] / scale);
+  const xEnd = Math.min(width, xVisibleDomain[1] / scale);
+  const yEnd = Math.min(height, yVisibleDomain[1] / scale);
+
+  const { tileSize } = api;
+  const tileOffsets = getTilesOffset(
+    new Vector2(xOrigin, yOrigin),
+    { width: xEnd - xOrigin, height: yEnd - yOrigin },
+    tileSize
+  );
+
+  // Sort tiles from closest to vis center to farthest away
+  const center = new Vector2(0.5 * (xOrigin + xEnd), 0.5 * (yOrigin + yEnd));
+  tileOffsets.sort((a, b) => {
+    return (
+      tileToCenterSquareDistance(a, center, tileSize) -
+      tileToCenterSquareDistance(b, center, tileSize)
+    );
+  });
```

Comment:
You could make a function `sortByDistanceFromCenter` to inline this (including the comment) but no pressure :smile: